.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,88 @@
+name: E2E Tests
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build-e2e-images:
+    name: Build & Run E2E Images
+    runs-on: [self-hosted, linux, X64, jammy, xlarge]
+    steps:
+      - name: Check out repo
+        uses: actions/checkout@v4
+      - name: Install requirements
+        run: |
+          sudo apt install make
+          sudo snap install go --classic
```

Comment:
Let's use an action to be consistent with the rest of the code base and ensure we know which version is installed.

```
- name: Install Go
  uses: actions/setup-go@v5
  with:
    go-version: "1.22"
```

Reply:
I ended up just setting the `channel` for the snap. For some reason the go binary was not available in follow-up steps and I don't think it is worth to dig into that right now.